junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/ExtensionValuesStore.java

```diff
@@ -110,7 +110,16 @@ Object remove(Namespace namespace, Object key) {
 
 	private StoredValue getStoredValue(Namespace namespace, Object key) {
 		CompositeKey compositeKey = new CompositeKey(namespace, key);
-		return this.storedValues.get(compositeKey);
+
+		/*
+		 * This method can be called from another ExtensionValuesStore instance,
+		 * so we must synchronized on the monitor.
```

Comment:
It _can_ be, but is there currently anywhere in the code base where that occurs from outside a synchronized block?